reconcile/openshift_tekton_resources.py

```diff
@@ -0,0 +1,360 @@
+from reconcile import openshift_resources_base as orb
+from reconcile.status import ExitCodes
+from reconcile.utils import threaded, gql
+from reconcile.utils.oc import OC_Map, StatusCodeError
+from reconcile.utils.defer import defer
+from reconcile.utils.openshift_resource import OpenshiftResource, \
+                                               ResourceInventory
+from reconcile.utils.semver_helper import make_semver
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'openshift-tekton-resources'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+OBJECTS_PREFIX = 'otr'
+RESOURCE_MAX_LENGTH = 63
+
+# Defaults
+DEFAULT_DEPLOY_RESOURCES = {'requests': {'cpu': '50m',
+                                         'memory': '200Mi'},
+                            'limits': {'cpu': '200m',
+                                       'memory': '300Mi'}}
+# Type alias
+SaasFile = Dict[str, Any]
+TektonNamespace = Dict[str, Any]
+LoadedYamlResource = Dict[str, Any]
+
+SAAS_FILES_QUERY = """
+{
+  saas_files: saas_files_v2 {
+    path
+    name
+    configurableResources
+    pipelinesProvider {
+      name
+      provider
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
+      }
+    }
+  }
+}
+"""
+
+
+class OpenshiftTektonResourcesNameTooLong(Exception):
+    pass
+
+
+class OpenshiftTektonResources:
+    """Integration runner class"""
+    def __init__(self,
+                 dry_run: bool,
+                 thread_pool_size: int,
+                 internal: Optional[bool],
+                 use_jump_host: bool,
+                 saas_file_name: Optional[str]) -> None:
+        self.dry_run = dry_run
+        self.thread_pool_size = thread_pool_size
+        self.internal = internal
+        self.use_jump_host = use_jump_host
+        self.saas_file_name = saas_file_name
+        self.ri = ResourceInventory()
+        self.gqlapi = gql.get_api()
+
+    def run(self) -> bool:
+        """Runs the integration"""
+        saas_files: List[SaasFile] = self._get_saas_files()
+        if not saas_files:
+            LOG.info("No saas files found to be processed")
+            return False
+
+        tkn_providers = self._get_tkn_providers(saas_files)
+        tkn_namespaces = [pp['namespace'] for pp in tkn_providers.values()]
+
+        oc_map = OC_Map(
+            namespaces=tkn_namespaces,
+            integration=QONTRACT_INTEGRATION,
+            settings=queries.get_app_interface_settings(),
+            internal=self.internal,
+            use_jump_host=self.use_jump_host,
+            thread_pool_size=self.thread_pool_size)
+        defer(lambda: oc_map.cleanup())
+
+        # current state
+        state_specs = ob.init_specs_to_fetch(
+            self.ri, oc_map, namespaces=tkn_namespaces,
+            override_managed_types=['Pipeline', 'Task'])
+
+        threaded.run(self.fetch_current_resource, state_specs,
+                     self.thread_pool_size)
```

Comment:
this whole section together with `fetch_current_resource` seem to be an exact copy of openshift_base.fetch_current_state?

Reply:
yes, it is very rough... I just wanted something to prove the idea we had discussed. The fetch current state part is something I just wanted to work as the main thing was in building the desired state from the pipelines provider and the saas files. I pushed it to save it but It wasn't really meant for review, sorry.

Reply:
But since you ask, I think I took that code mainly from the terraform resources integration :)

Reply:
terraform-resources predates openshift-base IMO :)

Reply:
i will use `fetch_current_resource` from openshift_base